src/main/java/com/commercetools/project/sync/Syncer.java

```diff
@@ -47,38 +53,66 @@
     B extends BaseSync<S, U, V>> {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(Syncer.class);
+
+  private final B sync;
   private final SphereClient sourceClient;
   private final SphereClient targetClient;
-  private final B sync;
-  private final C query;
+  private final CustomObjectService customObjectService;
 
+  /**
+   * Instantiates a {@link Syncer} which is used to sync resources from a source to a target
+   * commercetools project.
+   *
+   * @param sync The sync module that is used for syncing the transformed drafts to the target
+   *     project.
+   * @param sourceClient the client used for querying data from the source commercetools project.
+   * @param targetClient the client used for syncing the transformed drafts into the target
+   *     commercetools project.
+   * @param customObjectService service that is used for fetching and persisting the last sync
+   *     timestamp for delta syncing.
+   */
   public Syncer(
       @Nonnull final B sync,
-      @Nonnull final C query,
       @Nonnull final SphereClient sourceClient,
-      @Nonnull final SphereClient targetClient) {
+      @Nonnull final SphereClient targetClient,
+      @Nonnull final CustomObjectService customObjectService) {
     this.sync = sync;
-    this.query = query;
     this.sourceClient = sourceClient;
     this.targetClient = targetClient;
+    this.customObjectService = customObjectService;
   }
 
   /**
-   * Fetches the {@code CTP_SOURCE_CLIENT} project resources of type {@code T} with all needed
-   * references expanded and treats each page as a batch to the sync process. Then executes the sync
-   * process of all pages in parallel. It then returns a completion stage containing no result after
-   * the execution of the sync process and logging the result.
+   * Fetches the sourceClient's project resources of type {@code T} with all needed references
+   * expanded and treats each page as a batch to the sync process. Then executes the sync process of
+   * all pages in parallel. It then returns a completion stage containing no result after the
+   * execution of the sync process and logging the result.
+   *
+   * <p>Note: The method checks if there was a last sync time stamp persisted as a custom object in
```

Comment:
I think this note is not necessary if it is not optional :)

Reply:
yeah, actually still this needs to be changed because it will be optional. The documentation will be updated/added/improved in a separate PR.

Reply:
good 👍